lib/jenkins_pipeline_builder/compiler.rb

```diff
@@ -44,9 +44,15 @@ def get_settings_bag(item_bag, settings_bag = {})
       my_settings_bag.merge(bag)
     end
 
+    def do_compile(item, settings = {})
```

Comment:
It purely exists to abstract the new recursive compilation structure - it hides it from higher level classes to avoid having to refactor everything at this time.  Is there a better name that you can think of?